api/routes/device_test.go

```diff
@@ -32,17 +91,647 @@ func TestUpdatePendingStatus(t *testing.T) {
 		req, _ := http.NewRequest(http.MethodPatch, "/devices/:uid/:status", bytes.NewBuffer([]byte{}))
 		req.Header.Set("Content-Role", "application/json")
 		req.Header.Set("X-Role", guard.RoleOwner)
+
 		echoContext := e.NewContext(req, rec)
 		echoContext.SetParamNames("uid", "status")
 		echoContext.SetParamValues("123", "pending")
+
 		mock.On("UpdatePendingStatus", ctx, models.UID("123"), models.DeviceStatus("pending"), "").Return(svc.ErrDeviceLimit)
 
 		apictx := gateway.NewContext(mock, echoContext)
-
 		output := h.UpdatePendingStatus(*apictx)
 		err, ok := output.(errors.Error)
-		assert.True(t, ok)
 
+		assert.True(t, ok)
```

Comment:
The assert should be in the previous code block, as it ensures the casting has succeeded.

Reply:
resolved during refactoring